### PR TITLE
Document AWS_PROFILE support

### LINKED ISSUE #3506: enhance docs to cover support for AWS_PROFILE
Output of `restic version`
--------------------------
restic 0.12.1 compiled with go1.16.6 on linux/amd64

How did you run restic exactly?
-------------------------------
n/a

What backend/server/service did you use to store the repository?
----------------------------------------------------------------
n/a

Expected behavior
-----------------
n/a

Actual behavior
---------------
n/a

Steps to reproduce the behavior
-------------------------------
n/a

Do you have any idea what may have caused this?
-----------------------------------------------
n/a

Do you have an idea how to solve the issue?
-------------------------------------------
Add a line to the doc/040_backup.rst file.  I will submit PR.

Did restic help you today? Did it make you happy in any way?
------------------------------------------------------------
Yes!

### PR DESCRIPTION
**Pull request by ahmgithubahm**
What does this PR change? What problem does it solve?
-----------------------------------------------------
Since restic (or whatever library it is using) seems to respect/use AWS_PROFILE, it's worth documenting this.

Was the change previously discussed in an issue or on the forum?
----------------------------------------------------------------
I don't think so.

Checklist
---------

- [x] I have read the [contribution guidelines](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#providing-patches).
- [x] I have [enabled maintainer edits](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [x] I have added tests for all code changes.
- [x] I have added documentation for relevant changes (in the manual).
- [x] There's a new file in `changelog/unreleased/` that describes the changes for our users (see [template](https://github.com/restic/restic/blob/master/changelog/TEMPLATE)).
- [x] I have run `gofmt` on the code in all commits.
- [x] All commit messages are formatted in the same style as [the other commits in the repo](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#git-commits).
- [x] I'm done! This pull request is ready for review.

### CODE REVIEW

File: doc/040_backup.rst [hunk 1]
```diff
@@ -529,6 +529,7 @@ environment variables. The following lists these environment variables:
     AWS_ACCESS_KEY_ID                   Amazon S3 access key ID
     AWS_SECRET_ACCESS_KEY               Amazon S3 secret access key
     AWS_DEFAULT_REGION                  Amazon S3 default region
+    AWS_PROFILE                         Amazon credentials profile (can be used instead of the keys and region)
```
Comment (rawtaz) [inline doc/040_backup.rst:532]: Can you please do the following:

- Remove "of the" in your description for `AWS_PROFILE`.
- Add also the `AWS_SHARED_CREDENTIALS_FILE` one.

**Review by rawtaz — CHANGES_REQUESTED** [review]
I requested a few changes, can you please squash it all into one single commit after they're made? 👍

**Comment by MichaelEischer** [comment]
@ahmgithubahm Any progress here?

**Review by rawtaz — APPROVED** [review]
Thank you!

**Pull request merged**